Mathlib/Geometry/Manifold/ContMDiff/Product.lean

```diff
@@ -347,3 +347,133 @@ theorem contMDiff_pi_space :
 @[deprecated (since := "2024-11-20")] alias smooth_pi_space := contMDiff_pi_space
 
 end PiSpace
+
+section disjointUnion
+
+variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {n : WithTop ℕ∞}
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E'] {H' : Type*} [TopologicalSpace H']
+  {J : Type*} {J : ModelWithCorners 𝕜 E' H'}
+  {N N' : Type*} [TopologicalSpace N] [TopologicalSpace N'] [ChartedSpace H' N] [ChartedSpace H' N']
+
+open Topology
+
+lemma ContMDiff.inl : ContMDiff I I n (@Sum.inl M M') := by
+  intro x
+  rw [contMDiffAt_iff]
+  refine ⟨continuous_inl.continuousAt, ?_⟩
+  -- In extended charts, .inl equals the identity (on the chart sources).
+  apply contDiffWithinAt_id.congr_of_eventuallyEq; swap
+  · simp [ChartedSpace.sum_chartAt_inl]
+    congr
+    apply Sum.inl_injective.extend_apply (chartAt _ x)
+  set C := chartAt H x with hC
+  have : I.symm ⁻¹' C.target ∩ range I ∈ 𝓝[range I] (extChartAt I x) x := by
+    rw [← I.image_eq (chartAt H x).target]
+    exact (chartAt H x).extend_image_target_mem_nhds (mem_chart_source _ x)
+  filter_upwards [this] with y hy
+  simp [extChartAt, sum_chartAt_inl, ← hC, Sum.inl_injective.extend_apply C, C.right_inv hy.1,
+    I.right_inv hy.2]
+
+lemma ContMDiff.inr : ContMDiff I I n (@Sum.inr M M') := by
+  intro x
+  rw [contMDiffAt_iff]
+  refine ⟨continuous_inr.continuousAt, ?_⟩
+  -- In extended charts, .inl equals the identity (on the chart sources).
+  apply contDiffWithinAt_id.congr_of_eventuallyEq; swap
+  · simp [ChartedSpace.sum_chartAt_inr]
+    congr
+    apply Sum.inr_injective.extend_apply (chartAt _ x)
+  set C := chartAt H x with hC
+  have : I.symm ⁻¹' (chartAt H x).target ∩ range I ∈ 𝓝[range I] (extChartAt I x) x := by
+    rw [← I.image_eq (chartAt H x).target]
+    exact (chartAt H x).extend_image_target_mem_nhds (mem_chart_source _ x)
+  filter_upwards [this] with y hy
+  simp [extChartAt, sum_chartAt_inr, ← hC, Sum.inr_injective.extend_apply C, C.right_inv hy.1,
+    I.right_inv hy.2]
+
+lemma extChartAt_inl_apply {x : M} :
+    (extChartAt I (.inl x : M ⊕ M')) (Sum.inl x) = (extChartAt I x) x := by
```

Comment:
```suggestion
lemma extChartAt_inl_apply {x y : M} :
    (extChartAt I (.inl x : M ⊕ M')) (Sum.inl y) = (extChartAt I x) y := by
```
(untested). If this works, mark this as a simp lemma? Same thing for the next lemma.

Reply:
Indeed, this works - marked as simp. Thanks for suggesting this.

Reply:
In fact is should be unsimped, because simp can prove this -- and the proof should just be `by simp`.

Reply:
Indeed, as the lemma says. Done.